pkg/southbound/topocache/topocache.go

```diff
@@ -85,3 +81,27 @@ func LoadDeviceStore(file string, topoChannel chan<- events.TopoEvent) (*DeviceS
 
 	return &deviceStore, nil
 }
+
+// AddDevice adds the specified device to the device inventory
+func (store *DeviceStore) AddDevice(id string, device Device) error {
+	err := validateDevice(id, device)
+	if err == nil {
+		store.Store[id] = device
```

Comment:
It might be worth checking that this id does not exist already - otherwise the meaning here is AddOrOverwrite